src/lambda_codebase/initial_commit/bootstrap_repository/adf-build/shared/helpers/terraform/get_accounts.py

```diff
@@ -90,6 +97,45 @@ def get_accounts():
     )
 
 
+def get_accounts_from_tags():
+    accounts = get_accounts()
+    tag_filters = []
+    for tag in TARGET_TAGS.split(','):
+        tag_name = tag.split('=')[0]
+        tag_value = tag.split('=')[1]
+        tag_filters.append({
+            "Key": tag_name,
+            "Value": tag_value})
+
+    LOGGER.info(
+        "Tag filters %s",
+        tag_filters
+    )
+
+    organizations = get_boto3_client(
+        'organizations',
+        (
+            f'arn:{PARTITION}:sts::{MANAGEMENT_ACCOUNT_ID}:role/'
+            f'{CROSS_ACCOUNT_ACCESS_ROLE}-readonly'
+        ),
+        'getaccountIDs',
+    )
+    filtered_accounts = []
+    for account in accounts:
+        tags = account['Tags'] = organizations.list_tags_for_resource(
```

Comment:
Instead of `list_tags_for_resource` and filtering here, you could also rely on the implementation that ADF uses for CodePipeline targets:

* https://github.com/awslabs/aws-deployment-framework/blob/f2bb3471c1c56a8fd42a33a284f50e98ba47a12e/src/lambda_codebase/initial_commit/bootstrap_repository/adf-build/shared/python/organizations.py#L423-L429
* https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/resourcegroupstaggingapi/client/get_resources.html

Ideally, the way the targets are defined would match the same style as defined in the deployment map schema.
Which also allows a single tag with multiple values. Where the tag key should match any of that list.